repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -206,6 +222,74 @@ object Evaluator{
       )
     }
 
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
+      _ <- Catching{ case e: ThreadDeath => interrupted() }
+      (classFiles, importData) <- compileClass(
+        s"""
```

Comment:
As I see they only have the `Catching` line in common. Should I break up the `evalClass` call in `processLine` to separate `compileClass` and `loadClass` calls?